### PR TITLE
Add Spigot support, and non-zip URLs

### LINKED ISSUE #50: [mc] Allow given WORLD URLs to not end in .zip
@sshipway I was testing out my addition of `WORLD` to my container-deployer web app and noticed that download links from http://www.minecraftmaps.com/ are exposed as something like http://www.minecraftmaps.com/parkour-maps/aspect-parkour/download but then do a temp redirect to the actual zip's URL.

Since `wget` can follow redirects, perhaps we should let it download and then verify the final the filename/content.

### PR DESCRIPTION
**Pull request by sshipway**
- Allow world URLs to end in something other than .zip (fixes #50 )
- Some whitespace fixes (tabs->4space)
- Allow TYPE=SPIGOT and TYPE=BUKKIT.  Note that only version 1.8.8 is currently available.
- Rename directories for dimensions if TYPE=SPIGOT/BUKKIT
- Some text changes to reflect new features
- Allow type options to be lower-case
- Add documentation
- Add /plugins mount for external bukkit plugins (like /mods for forge mods)

### CODE REVIEW

File: minecraft-server/start-minecraft.sh [hunk 1]
```diff
@@ -33,19 +33,35 @@ esac
...
-  VANILLA)
-    SERVER="minecraft_server.$VANILLA_VERSION.jar"
-
-    if [ ! -e $SERVER ]; then
-      echo "Downloading $SERVER ..."
-      wget -q https://s3.amazonaws.com/Minecraft.Download/versions/$VANILLA_VERSION/$SERVER
-    fi
-  ;;
+  *BUKKIT|*bukkit|SPIGOT|spigot)
+    TYPE=SPIGOT
```
Comment (itzg) [inline minecraft-server/start-minecraft.sh]: It seems like this line would effectively coerce BUKKIT type into SPIGOT.

File: minecraft-server/start-minecraft.sh [hunk 2]
```diff
@@ -33,19 +33,35 @@ esac
...
+    case $VANILLA_VERSION in
+      1.8*)
+        URL=/spigot18/$SERVER
+        ;;
+      *)
+        echo "That version of $SERVER is not available."
+        exit 1
+      ;;
+    esac
+    wget -q https://getspigot.org$URL
```
Comment (itzg) [inline minecraft-server/start-minecraft.sh]: Awesome! I had only come across their build-it-yourself instructions before. I'm glad you knew about them having a pre-built jar available.

File: minecraft-server/start-minecraft.sh (hunk 1 above)
Comment (sshipway) [inline minecraft-server/start-minecraft.sh]: Actually, Bukkit is currently only available as a 100% compatible compile from the Spigot people (see how $SERVER is changed) due to DMCA issues.  The SPigot API is an extension of the Bukkit one.

Comment (sshipway) [inline minecraft-server/start-minecraft.sh]: Both come from the same place and require the same modifications to the world structure, so are handled together.

**Comment by sshipway** [comment]
I'm using the TYPE=SPIGOT to later ensure the correct handling of plugins and world structure; TYPE=BUKKIT is handled here to download the correct craftbukkit-server.jar file and it is set to TYPE=SPIGOT to ensure the correct handling later.

File: minecraft-server/start-minecraft.sh (hunk 1 above)
Comment (itzg) [inline minecraft-server/start-minecraft.sh]: Got it. I remember there was controversy with Bukkit...and then I hadn't researched much of Spigot. Thanks.

File: minecraft-server/start-minecraft.sh (hunk 2 above)
Comment (sshipway) [inline minecraft-server/start-minecraft.sh]: The only pity is that there is no simple way to get Version $X or a list of available versions, though you can get Latest 1.8.  As it is, only v1.8, 1.8.3 and 1.8.8 are available via the Spigot Jenkins and are all compatible.

File: minecraft-server/start-minecraft.sh [hunk 3]
```diff
@@ -108,6 +139,12 @@ case "X$WORLD" in
         fi
       done
     fi
+    if [ "$TYPE" = "SPIGOT" ]; then
```
Comment (sshipway) [inline minecraft-server/start-minecraft.sh]: This catches BUKKIT types as well as we set TYPE=SPIGOT for both cases

File: minecraft-server/start-minecraft.sh [hunk 4]
```diff
@@ -252,6 +289,11 @@ do
     cp -rf "$c" /data/config
   fi
 done
-
+if [ "$TYPE" = "SPIGOT" ]; then
```
Comment (sshipway) [inline minecraft-server/start-minecraft.sh]: Again, this catches BUKKIT types as well.  We use a separate volume for plugins as they are not compatible with mods though they look like them

**Pull request merged**